development/compiling/compiling_with_mono.rst

```diff
@@ -102,24 +107,83 @@ Example (X11)
     scons p=x11 target=debug tools=no module_mono_enabled=yes
     scons p=x11 target=release tools=no module_mono_enabled=yes
 
-If everything went well, apart from the normal output, SCons should have also built the *GodotSharpTools.dll* assembly and copied it together with the mono runtime shared library to the ``bin`` subdirectory.
+.. _compiling_with_mono_data_directory:
+
+Data directory
+--------------
+
+Export templates
+^^^^^^^^^^^^^^^^
+
+The name of the data directory for a export template differs based on the configuration it was built with. The format is ``data.mono.<platform>.<bits>.<target>``, e.g. ``data.mono.x11.32.debug`` or ``data.mono.windows.64.release``.
+
+In the case of export templates the data directory only contains Mono framework assemblies and configuration files, as well as some shared library dependencies like ``MonoPosixHelper``.
+
+In the case of macOS, where the export template is compressed as a zip file, the contents of the data directory can be placed in the following locations inside the zip:
+
++-------------------------------------------------------+---------------------------------------------------------------+
+| ``bin/data.mono.<platform>.<bits>.<target>/Mono/lib`` | ``/osx_template.app/Contents/Frameworks/GodotSharp/Mono/lib`` |
++-------------------------------------------------------+---------------------------------------------------------------+
+| ``bin/data.mono.<platform>.<bits>.<target>/Mono/etc`` | ``/osx_template.app/Contents/Resources/GodotSharp/Mono/etc``  |
++-------------------------------------------------------+---------------------------------------------------------------+
+
+Editor
+^^^^^^^^
+
+The name of the data directory for the Godot editor will always be ``GodotSharp``. The main structure of this directory has the following subdirectories:
+
+- ``Api`` (optional)
+- ``Mono`` (optional)
+- ``Tools`` (required)
+
+The ``Tools`` subdirectory contains tools required by the editor, like the ``GodotSharpTools`` assembly.
+
+The ``Mono`` subdirectory is optional. It can be used to bundle the Mono framework assemblies and configuration files with the Godot editor, as well as some shared library dependencies like ``MonoPosixHelper``. This is important to avoid issues that might arise when the installed Mono version in the user's system may not be the same as the one the Godot editor was built with. You can make SCons copy these files to this subdirectory by passing the option ``copy_mono_root=yes`` when building the editor.
+
+The ``Api`` directory is also optional. Godot API assemblies are not bundled with the editor by default. Instead the Godot editor will generate and build them on the user's machine the first time they are required. This can be avoiding by generating and building them manually and placing them in this subdirectory. If the editor can find them there, it will avoid the step of generating and building them again.
```

Comment:
type near "this can be avoiding" -> avoided*

Reply:
Fixed, thanks!